### PR TITLE
GO-2734 Add global name to cache

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-2734/add-cache-for-client-to-resolve-id-name

GlobalName detail is set to all participant objects in space.
Global name for local profile is retrieved from Subscription status that is requested on application start.
Global names of invited participants are retieved from Naming Service by AnyID and stored in cache.

### CODE REVIEW

File: core/payments/payments.go [hunk 1]
```diff
@@ -189,6 +200,24 @@ func (s *service) GetSubscriptionStatus(ctx context.Context, req *pb.RpcPayments
...
+	spc, err := s.spaceService.Get(context.Background(), s.account.PersonalSpaceID())
+	if err != nil {
+		log.Error("failed to get personal space id:" + err.Error())
+		return &out, nil
+	}
+	if err = spc.Do(s.account.MyParticipantId(s.account.PersonalSpaceID()), func(sb smartblock.SmartBlock) error {
+		if ds, ok := sb.(basic.DetailsSettable); ok {
+			return ds.SetDetails(nil, []*pb.RpcObjectSetDetailsDetail{{
+				Key:   bundle.RelationKeyGlobalName.String(),
+				Value: pbtypes.String(status.RequestedAnyName),
```
Comment (AnthonyAkentiev) [inline core/payments/payments.go:213]: maybe some check for the status of the subscription is required here, but leave it as-is
but not sure
and I will fix it by myself later, if needed

**Review by AnthonyAkentiev — APPROVED** [review]

File: core/identity/identity.go [hunk 2]
```diff
@@ -498,6 +519,21 @@ func (s *service) findProfile(identityData *identityrepoproto.DataWithIdentity)
 	return profile, rawProfile, nil
 }
 
+func (s *service) fetchIdentitiesGlobalNames(anyIDs []string) error {
```
[the remark this answers is not in the document]
Comment (KirillSto) [reply core/identity/identity.go]: Fixed

File: core/identity/identity.go [hunk 3]
```diff
@@ -392,6 +402,13 @@ func (s *service) observeIdentities(ctx context.Context) error {
 		return fmt.Errorf("failed to pull identity: %w", err)
 	}
 
+	if s.identityGlobalNames == nil && len(identities) != 0 {
```
[the remark this answers is not in the document]
Comment (KirillSto) [reply core/identity/identity.go]: Done

File: core/identity/identity.go [hunk 4]
```diff
@@ -498,6 +519,21 @@ func (s *service) findProfile(identityData *identityrepoproto.DataWithIdentity)
...
+	response, err := s.namingService.BatchGetNameByAnyId(context.Background(), &nameserviceproto.BatchNameByAnyIdRequest{AnyAddresses: anyIDs})
+	if err != nil {
+		return err
+	}
+	if response == nil {
+		return nil
+	}
+	s.identityGlobalNames = make(map[string]string, len(anyIDs))
+	for i, anyID := range anyIDs {
+		s.identityGlobalNames[anyID] = response.Results[i].Name
```
[the remark this answers is not in the document]
Comment (KirillSto) [reply core/identity/identity.go:532]: Check added

File: core/identity/identity_test.go [hunk 5]
```diff
@@ -33,13 +36,17 @@ const testObserverPeriod = 1 * time.Millisecond
...
 
 	identityRepoClient := newInMemoryIdentityRepo()
 	objectStore := objectstore.NewStoreFixture(t)
 	accountService := mock_account.NewMockService(t)
 	spaceService := mock_space.NewMockService(t)
 	fileAclService := mock_fileacl.NewMockService(t)
 	dataStore := datastore.NewInMemory()
+	wallet := mock_wallet.NewMockWallet(t)
+	nsClient := mock_nameserviceclient.NewMockAnyNsClientService(ctrl)
+	nsClient.EXPECT().BatchGetNameByAnyId(gomock.Any(), gomock.Any()).AnyTimes()
```
[the remark this answers is not in the document]
Comment (KirillSto) [reply core/identity/identity_test.go:49]: We expect BatchGetNameByAnyId to be called as global names are fetched from NS in observingLoop

**Review by deff7 — APPROVED** [review]

**Pull request merged**